afs-cassandra/src/main/java/com/powsybl/afs/cassandra/CassandraAppStorage.java

```diff
@@ -995,9 +996,25 @@ public void write(int b) throws IOException {
 
         @Override
         public void write(byte[] b, int off, int len) throws IOException {
-            gzos.write(b, off, len);
-            count += len;
-            executeIfNecessary();
+            if (len > config.getBinaryDataChunkSize()) {
+                int chunkOffset = off;
+                int chunkCount = 0;
+                while (chunkCount * config.getBinaryDataChunkSize() <= len) {
+                    int chunkLen = config.getBinaryDataChunkSize();
+                    if (len - chunkOffset < config.getBinaryDataChunkSize()) {
+                        chunkLen = len - chunkOffset;
+                    }
+                    gzos.write(Arrays.copyOfRange(b, chunkOffset, chunkOffset + chunkLen));
```

Comment:
Not sure that is necessary to copy the byte array. Maybe you can use the `gzos.write(bytes, offset, length)`